### DISCUSSION TITLE
Help with PCE PCB

### DISCUSSION THREAD

**Discussion by pmgducati** [darthcloud/BlueRetro#165] [Q&A] [answered]
Hello,
I have had a great time making cables for my consoles.  So far I have made Gamecube, 3DO, NES, and Genesis with no issues.  Everything works perfectly!  I recently created a PCB and cable for the PC Engine and am stumped.  The controls are not working as intended.  I am using the same main PCB with the ESP32 for all the other systems and it works fine.  See Below for the  buttons pressed vs output: 

R2 --> D Pad Up + I
A --> D Pad Right + II
Start --> D Pad Left + Run
Select --> D Pad Down + Select

For reference I am using all the parts spec'd in the BOM in the cable making section, a NODEMCU ESP32 and 8bitdo SN30pro+ controllers.  I attached gerbers and PDFs of the schematics and board files

Any help would be greatly appreciated

[PCE.pdf](https://github.com/darthcloud/BlueRetro/files/8055134/PCE.pdf)
[Main PCB.pdf](https://github.com/darthcloud/BlueRetro/files/8055135/Main.PCB.pdf)
[PCE_2022-01-25.zip](https://github.com/darthcloud/BlueRetro/files/8055136/PCE_2022-01-25.zip)
[Main_2022-01-25.zip](https://github.com/darthcloud/BlueRetro/files/8055137/Main_2022-01-25.zip)

**Comment by darthcloud** [comment]
Can you try with FW 0.16 https://github.com/darthcloud/BlueRetro/releases/tag/v0.16

This is original version that added PCE support and probably the last one I tested PCE with.

I'll setup my PCE and try too.

**Reply by pmgducati** [reply]
Just tried FW 0.16, unfortunately I am getting the same results.

**Reply by darthcloud** [reply]
Can you list a few games? Best if they are CD-ROM, as I don't have a PCE flash card.

**Reply by pmgducati** [reply]
I tried Golden Axe and Rayxbander with both FW 0.16 and 1.22, the second game is the best example as you can see the ship move left and pause or shoot and move forward etc.

I also attached a video of a pad test rom using FW 0.16 that I found that shows the issues.

https://user-images.githubusercontent.com/20894227/153780439-c08248b0-bf2d-4a42-88a4-0c4b7a4fa656.mp4

**Comment by darthcloud** [comment, marked as answer]
Try this build:
[v1.2.2-pce_test.zip](https://github.com/darthcloud/BlueRetro/files/8063466/v1.2.2-pce_test.zip)

This only improve 2/3 btns ctrl mode, tested with Rayxbander III and Bomberman 94, do not see pause or random press.

I removed the /OE line handling to speed things up, some game might not like this?

Edit: Problem was schematic had /OE and SEL line inverted (now corrected)

**Reply by pmgducati** [reply]
I found the part of the issue, now I double and triple checked this.  I think you have two pins swapped in the cable layout.  As I was testing the firmware issue, I noticed that the double presses that I was seeing were the pinouts.  U_I - Up/I, R_II - Right_II, D_SL - Down/Select and L_RN - Left/Run.  After finding this issue in FW1.00/.016/1.2.2 and the latest 1.2.2_pce_test I thought to myself this has to be a HW issue.  All the pins are either Power, Ground or directly related to buttons with the exception of OL and SEL.  Since I could get all the button presses and power was working I swapped the pin and boom!  No more double presses!  I thought this was a oops on my part and kept looking from my Schematic to yours and everything matches, I used a volt meter to check the pins over and over all is good.  

![Screenshot 2022-02-14 150748](https://user-images.githubusercontent.com/20894227/153946657-52c6769e-bf82-42dc-9ea1-f144a6f50d4b.png)

Now on to the final part.  The I and II buttons are mapped to an odd configuration.  I is the Right Trigger and II is the A (Right) button.  If I map to the correct locations, II being B (Down) and I Being A (Right), everything works great for the PCE.  But the issue is now that the button changes that I made for the PCE break the other systems since I changed the mapping.  Is there any way around this?

**Reply by darthcloud** [reply]
Yes good catch the schematic is indeed wrong! Sorry about that!!

The mapping is to match the 6 buttons layout to the genesis/saturn 6 btns one, so in regular 2 btns 1 is on R.

But I guess having default ctrl type on face buttons is better than 6 buttons consistency across systems. I'll think about it.

Between the original 1.2.2 and the pce-test one, now that you wired correctly, do you see an improvement? Like no random press anymore?

**Reply by darthcloud** [reply]
I uploaded a corrected version of the schematic.

**Reply by pmgducati** [reply]
Awesome thanks for updating it!  I tested the new wiring with every version of FW that I had (.016,1.00,1.2.2 and 1.2.2pcetest) and all perform as intended with no issues.  Also thanks for looking into the button layout for the controller.  I looked into the 6 button controller game list and it is VERY short (see Below) in the large list of PCE/TG16 games.  It would be greatly appreciated if the buttons could be remapped.  Would it be possible to create a profile with II and I on the A/B Buttons and a second for the 6 button controller?

Advanced VG
Art of Fighting
Battlefield '94
Fatal Fury 2
Fatal Fury Special
Fire Pro Female Wrestling
Kabuki Itouryodan
Kakutou Haou Densetsu Algunos
Linda Cube
Majhong Sword Princess Quest Gaiden
Martial Champions
Princess Maker 2
Street Fighter II'
Strip Fighter
Super Real Mahjong P ll + lll Custom
World Heroes 2 
Ys IV

**Reply by darthcloud** [reply]
Ok good to know, I think altered my cable timing somehow, I get random press without pce-test FW. What pull-up resistor are you using? 4K7 or something lower?

**Reply by pmgducati** [reply]
I was getting random presses when SEL and OE were swapped.  Once I fixed that the random presses stopped along with the double presses.  I am using 47K resistors as you spec'd in the diagram.  I spent an hour this morning testing (playing games hahaha).  I played Devils Crush, Bonk III, Raden, Golden Axe and Rayxbander with no issues at all.

**Reply by darthcloud** [reply]
ok sound good! enjoy!, probably something got loose in mine!

**Comment by darthcloud** [comment]
@pmgducati Someone want to buy some of your AOI PCBs, any way they can contact you other than GitHub?

**Comment by pmgducati** [comment]
They can reach out to me at ***@***.***

**Reply by pmgducati** [reply]
ok that didnt work, reach out to my username @ gmail
